### ISSUE TITLE
Protractor $timeout issue

### ISSUE THREAD

**Issue by fatbeard2** [lynndylanhurley/ng-token-auth#182]
I have a simple spec for sign up form. It looks something like this

```javascript
element(by.model('signUp.email')).sendKeys('some@email.com')
element(by.model('signUp.password')).sendKeys('password')
element(by.model('signUp.password_confirmation')).sendKeys('password')
element(by.css('button[type="submit"]')).click() //submit form
expect(browser.getCurrentUrl()).toMatch(/redirect_url/)
```

Protractor fails to run assertion because of timeout error: `Timed out waiting for Protractor to synchronize with the page after 11 seconds`. [Docs](https://github.com/angular/protractor/blob/master/docs/timeouts.md) for protractor says that 

> Before performing any action, Protractor asks Angular to wait until the page is synchronized. This means that all timeouts and http requests are finished. If your application continuously polls $timeout or $http, it will never be registered as completely loaded. You should use the $interval service (interval.js) for anything that polls continuously (introduced in Angular 1.2rc3).

After some investigation, I found out that the problem is [here](https://github.com/lynndylanhurley/ng-token-auth/blob/master/src/ng-token-auth.coffee#L640) This timeout is set to be resolved in future and protractor waits for it. I created a [fork](https://github.com/fatbeard2/ng-token-auth/tree/fix_protractor_timeout_error) where I changed $timeout service to $interval. 

Did anyone else had those issues? I didn't find any e2e tests for ng-auth-token, only `protractor.conf` file. Should I create PR for this fix?

**Comment by booleanbetrayal** [comment]
I'm somewhat reluctant to bend the code to fix a deficiency in the test tool. At a minimum, since $interval will run indefinitely, the callback function should also cancel the timer.

**Comment by askobara** [comment]
@booleanbetrayal, but if not fix this it will be create problems with a protractor tests
For avoiding indefinitely run of $interval @fatbeard2 can set [third parameter](https://code.angularjs.org/1.4.4/docs/api/ng/service/$interval) to 1, for example

**Comment by fatbeard2** [comment]
I switched to `capybara` with `capybara-angular` gem. And I'm having the same issue (`timeout while waiting for angular (Timeout::Error)`) if `ng-auth-token` is using `$timeout` I don`t think that this is a deficiency in the test tool. It is just a way how angular tells world - hey I'm testable now. And using $timeout prevents angular from doing so. I updated my fork to take advantage of `count` interval parameter.
